### DISCUSSION TITLE
ESP-Now - do you think this is possible?

### DISCUSSION THREAD

**Discussion by trizzosk** [nplan/HomeButtons#52] [Ideas]
Hello,

I do have several Home Buttons at my apartment to handle several different scenarios (lighting, AC's, etc.). Unfortunately I do have a "blind" spot where I cannot add any Wifi so I would like to utilize ESP-Now. I will deploy a simple ESP-Now to Wifi gateway (2 Wemos Mini something -> basically 2 ESP's connected together, running ESP-Now on one ESP and the other with usual Wifi with e.g. MQTT client). Did you consider this as an option? I know that the ESP is pretty universal so if I fork the code and replace usual Wifi with ESP-Now -> it shall work... I only have some doubts that the code (I haven't checked it) could be a bit complex.... 

SO now the question - do you have any plans to make an ESP-Now code or I have to burn my fingers and recode it by myself? 🤣 🤣 🤣 

Thx 👍

**Comment by nplan** [comment]
Hi,

I don't have plans to implement ESP-NOW at the moment. It might be possible to create an ESP-NOW to Wi-Fi MQTT bridge with another ESP32 and connect all Home Buttons to ESP-NOW mesh.

Though it would require a complete rewrite of the networking code.

Might be simpler to add a Wi-Fi extender... 😉
